src/core/validation/validators/UrlValidator.ts

```diff
@@ -25,11 +25,7 @@ export class UrlValidator {
    */
   public static validate(value: any): boolean {
     const pattern = new RegExp(
-      '^(https?:\\/\\/)?' + // protocol
-      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.?)+[a-z]{2,}|' + // domain name
-      '((\\d{1,3}\\.){3}\\d{1,3}))' + // ip (v4) address
-      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port
-        '(\\#[-a-z\\d_]*)?$',
```

Comment:
i would prefer to keep the above regular expression as it is well **documented**.

Reply:
I am sorry @vincent-lee90 I didnt see the Issue about this but it seems this regular expression is the cause of performance issues.

Please re-introduce your regular expression and keep the unit tests above :+1: